README.md

```diff
@@ -0,0 +1,87 @@
+# multiplex-reservation
+
+## Table of contents
+* [Informacje Ogólne](#informacje-ogólne)
+* [Komunikacja z aplikacją](#komunikacja-z-aplikacją)
+* [Warstwa Service](#warstwa-service)
+* [Generowanie Kina](#generowanie-kina)
+* [Odnośnie wymagań](#odnośnie-wymagań)
+
+## Informacje Ogólne
+
+Aplikacja webowa napisana w języku Java z użyciem frameworka Spring umozliwuająca rezerwację miejsc w kinie typu multiplex.
+Aplikacja nie posiada zaimplementowanego frontendu. Komunikowałem się z nią i testowałem za pomocą aplikacji Postman.
+Aplikacja korzysta z JPA oraz bazy danych H2 pod URL *"http://localhost:8080/console"* z adresem URL JDBC *"jdbc:h2:file:./bazaDanych"*.
+
+Baza danych zbudowana jest za pomocą 6 encji: User, Reservation, Screening, Movie, CinemaRoom, Seat.
+Każda z nich posiada swoje prywatne atrybuty oraz jest połączona z odpowiednimi innymi encjami.
+Każda encja korzysta z Repozytorium, które implementuje *PagingAndSortingRepository*.
+
+## Komunikacja z aplikacją
+
+Apliakcja obsługuje REST API i korzysta z dwóch komunikatów *GET* i *POST* w wielu wariantach:
+
+* Wypisanie pokazów filmowych konkretnego dnia w konkretnym przedziale czasowym za pomocą scieżki: 
+*/multiplex/reservation/screening/interval*
+
+Podać należy 3 paramentry w następującej kolejności: dzień, godzina *Od*, godzina *Do*.  
+Przykładowo: localhost:8080/multiplex/reservation/screening/interval?date=2&timeMin=10&timeMax=17  
+Aplikacja na wyjściu daje plik JSON wypisujący wszystkie dane na temat pokazów filmowych, które mieszczą się w podanym przedziale czasowym.
+
+* Wypisanie posortowanych alfabetycznie tytułów filmów w konkretnym przedziale czasowym za pomocą scieżki: 
+*/multiplex/reservation/screening/titles*
+
+Podać należy 3 paramentry w następującej kolejności: dzień, godzina *Od*, godzina *Do*.  
+Przykładowo: localhost:8080/multiplex/reservation/screening/titles?date=2&timeMin=10&timeMax=17  
+Aplikacja na wyjściu wyświetla listę posorotwanych alfabetycznie tytułów filmów.
+
+* Wyświetlenie niezarezerwowanych miejsc dla konkretnego pokazu filmowego za pomocą ścieżki:
+*/multiplex/reservation/chooseseat*
+
+Należy podać jeden parametr będący Id pokazu filmowego (zostałby przesłany poprzez frontend)  
+Przykładowo: localhost:8080/multiplex/reservation/chooseseat?screeningId=1  
+Aplikacaja zwraca listę wolnych miejsc wraz z ich danymi w formacie JSON
+
+* Dodanie nowego użytkownika (właściciela rezerwacji) pod ścieżką:
+*/multiplex/reservation/screening/user*
+
+Należy podać 3 parametry: imię, nazwisko i email.  
+Przykładowo: localhost:8080/multiplex/reservation/user?name=test&surname=testtest&email=test@test.pl  
```

Comment:
Czy użycie HTTP GET to najlepszy pomysł? Czemu może to być złe? Masz pomysł jak to zrobić lepiej?

Reply:
Nie wiem czy dobrze rozumiem pytanie. Wydaje mi się, że używam HTTP POST.  :)

Reply:
Podajesz parametry w query stringu - stąd nasuwa mi się GET.